ftplugin/python/vim_ipython.py

```diff
@@ -172,17 +172,25 @@ def km_from_string(s=''):
         # 0.13
         kc = km
     kc.start_channels()
-    send = kc.shell_channel.execute
+
+    try:
+        send = kc.execute
+    except AttributeError:
+        # < 3.0
+        send = kc.shell_channel.execute
 
     #XXX: backwards compatibility for IPython < 0.13
-    import inspect
-    sc = kc.shell_channel
-    num_oinfo_args = len(inspect.getargspec(sc.object_info).args)
-    if num_oinfo_args == 2:
-        # patch the object_info method which used to only take one argument
-        klass = sc.__class__
-        klass._oinfo_orig = klass.object_info
-        klass.object_info = lambda s,x,y: s._oinfo_orig(x)
+    try:
+        import inspect
+        sc = kc.shell_channel
+        num_oinfo_args = len(inspect.getargspec(sc.object_info).args)
+        if num_oinfo_args == 2:
+            # patch the object_info method which used to only take one argument
+            klass = sc.__class__
+            klass._oinfo_orig = klass.object_info
+            klass.object_info = lambda s,x,y: s._oinfo_orig(x)
+    except:
+        pass
 
```

Comment:
not sure if just silently ignoring all errors is the right approach here, but in the interest of breathing some life back into vim-ipython so it moves faster than a :snail: - I'll merge this as is.